pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Agent.groovy

```diff
@@ -76,16 +78,29 @@ class Agent extends MappedClosure<Object,Agent> implements Serializable {
 
             DeclarativeAgent a = DeclarativeAgentDescriptor.instanceForDescriptor(foundDescriptor, argMap)
 
-            boolean doCheckout = false
+            boolean doCheckout = true
+            Map<String,DeclarativeOption> options = [:]
+            if (root?.options?.options) {
+                options.putAll(root?.options?.options)
+            }
+
             if (context instanceof Root) {
                 a.setInStage(false)
-            } else {
+            } else if (context instanceof Stage) {
                 a.setInStage(true)
+                if (((Stage)context).options?.options) {
+                    options.putAll(((Stage) context).options?.options)
+                }
             }
-            if (root != null) {
-                SkipDefaultCheckout skip = (SkipDefaultCheckout) root?.options?.options?.get("skipDefaultCheckout")
-                if (!skip?.isSkipDefaultCheckout()) {
-                    doCheckout = true
+            if (!options.isEmpty()) {
+                SkipDefaultCheckout skip = (SkipDefaultCheckout) options.get("skipDefaultCheckout")
+                if (skip?.isSkipDefaultCheckout()) {
+                    doCheckout = false
+                }
+
+                CheckoutToSubdirectory subdir = (CheckoutToSubdirectory) options.get("checkoutToSubdirectory")
```

Comment:
Is there validation in place to make sure someone doesn't do `skipDefaultCheckout` and `checkoutToSubdirectory` at the same time?

Reply:
No - not sure I want to add that level of validation, honestly. In practice, that'd just make the `checkoutToSubdirectory` meaningless, since the `skipDefaultCheckout` would override it.